### PR TITLE
ML : Kaggle's Titanic Prediction

### LINKED ISSUE #7860: ML : Kaggle's Titanic Prediction
### 📌 Issues for Week 9 - Basic Projects of Machine Learning

🔴 **Project Title** : Kaggle's Titanic Prediction
🔴 **Aim of the Project** :  Use machine learning to create a model that predicts which passengers survived the Titanic shipwreck.
🔴 **Dataset** : https://www.kaggle.com/c/titanic/data
🔴 **Domain** : Machine Learning
🔴 **Special Credit** : **Try to implement more than three machine learning algorithms and make a comparison between them, then conclude about the best fit algorithm to create the model. All the algorithms must be checked through the accuracy scores to determine the best fitted model**

Welcome to 'ML' Team, good to see you here, in this week we'll be creating several basic projects of this Machine Learning Domain. 

**************************************************************

### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.
* A Demo Project has been created to guide you about the whole structure of presenting the Project in this repository. Here's the link of the Demo Project - https://github.com/girlscript/winter-of-contributing/tree/Machine_Learning/Machine_Learning/Basic_Projects_of_Machine_Learning/Demo_Project

**Take the Reference from these two projects attached here,** 
1. https://github.com/prathimacode-hub/ML-ProjectKart/tree/main/Airbnb%20Price%20Prediction
2. https://github.com/prathimacode-hub/ML-ProjectKart/tree/main/Bike%20Crash%20Analysis

***********************************************************************

:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Changes should be made inside the `Machine Learning/Machine_Learning/Basic_Projects_of_Machine_Learning` Branch.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.
- This issue is only for 'GWOC' contributors of 'Machine Learning' domain.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- Batch Number : 
- GitHub Profile Link : 
- Approach for this Project :
- Are you a participant of GWoC21?
    - [ ] YES
    - [ ]  No

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

### Domain

Machine Learning

### Type of Contribution

Others

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ashish-reddy-20-08**
<hr>

## Description 📜

Worked on Titanic prediction project. 
Fixes #7860 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Machine Learning

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor.

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# Kaggle's Titanic Prediction
```
[the remark this answers is not in the document]
Comment (ashish-reddy-20-08) [reply Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md]: I have made the changes mam, can you please check

**Review by Isha307 — CHANGES_REQUESTED** [review]
In the Conclusion and Discussion section, please compare the accuracy of all models you have used and also upload the dataset(csv files) as they are very small in size.

**Comment by ashish-reddy-20-08** [comment]
> In the Conclusion and Discussion section, please compare the accuracy of all models you have used and also upload the dataset(csv files) as they are very small in size.

Done mam

File: Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+
+
+
+## Goal/Aim of the project :
+
+Goal: For each passenger, predict the label (0 or 1).
+
+## Libraries used :
```
Comment (abhisheks008) [inline Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md:30]: Provide the libraries you have used here under this section. @ashish-reddy-20-08

**Review by abhisheks008 — CHANGES_REQUESTED** [review]

File: Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md (hunk 2 above)
Comment (ashish-reddy-20-08) [reply Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md:30]: Done sir , i made the changes

Comment (abhisheks008) [reply Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md:30]: 1. Write only `sklearn`
2. Include all the libraries which are mentioned in the `requirements.txt` file of your project.

@ashish-reddy-20-08

**Review by abhisheks008 — CHANGES_REQUESTED** [review]

File: Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md (hunk 2 above)
Comment (abhisheks008) [reply Machine_Learning/Basic_Projects_of_Machine_Learning/Kaggle's Titanic Prediction/Model/README.md:30]: Approved! @ashish-reddy-20-08

**Review by abhisheks008 — APPROVED** [review]

**Review by Isha307 — APPROVED** [review]

**Pull request merged**